src/app/components/shared/formly/location-input.component.spec.ts

```diff
@@ -1 +1,170 @@
+   */
+  function explicitlySetMarket(longitude: number, latitude: number) {
+    spectator.component.updateModel(longitude, latitude);
+    spectator.detectChanges();
+  }
+
+  function assertMapModelCoordinates(
+    map: MockedComponent<MapComponent>,
+    longitude: number,
+    latitude: number
+  ) {
+    expect(map.markers.toArray()[0]["position"]["lng"]).toEqual(longitude);
+    expect(map.markers.toArray()[0]["position"]["lat"]).toEqual(latitude);
+  }
+
+  it("should create", () => {
+    expect(spectator.component).toBeInstanceOf(LocationInputComponent);
+  });
+
+  it("should display the position of the marker in input boxes", fakeAsync(() => {
+    const defaultLongitudeValue = modelData.longitude();
+    const defaultLatitudeValue = modelData.latitude();
+
+    explicitlySetMarket(defaultLongitudeValue, defaultLatitudeValue);
+
+    tick();
+
+    expect(getLongitudeInput().value).toEqual(defaultLongitudeValue.toString());
+    expect(getLatitudeInput().value).toEqual(defaultLatitudeValue.toString());
+  }));
+
+  it("should update the marker model if the location is updated through the input field/form", () => {
+    const map = spectator.query(mockMapComponent);
+    const defaultLongitudeValue = modelData.longitude();
+    const defaultLatitudeValue = modelData.latitude();
+
+    // set the markers to a starting position
+    explicitlySetMarket(defaultLongitudeValue, defaultLatitudeValue);
+
+    // change the markers location through the input field
+    const updatedLongitudeValue = modelData.longitude();
+    const updatedLatitudeValue = modelData.latitude();
+
+    updateMarkerThroughInput(updatedLongitudeValue, updatedLatitudeValue);
+
+    // assert that the marker location has changed to the new location
+    assertMapModelCoordinates(map, updatedLongitudeValue, updatedLatitudeValue);
+  });
+
+  // the event emitter that triggers these methods is tested in the baw-map component
+  it("should update the marker model if the marker is dragged", () => {
+    const map = spectator.query(mockMapComponent);
+    const defaultLongitudeValue = modelData.longitude();
+    const defaultLatitudeValue = modelData.latitude();
+
+    // spy on the updateModel method so that we can assert later that it was called
+    // with the correct parameters
+    explicitlySetMarket(defaultLongitudeValue, defaultLatitudeValue)
+    spyOn<any>(spectator.component, "updateModel").and.callThrough();
+
+    // simulate moving the marker to a new position
+    const updatedLongitude = modelData.longitude();
+    const updatedLatitude = modelData.latitude();
+
+    // send the move event
+    map.newLocation.emit({
+      domEvent: new Event("mapDragend"),
+      latLng: {
+        lng: () => updatedLongitude,
+        lat: () => updatedLatitude,
+        equals: () => null,
+        toJSON: () => null,
+        toUrlValue: () => null
+      },
+      stop: () => {}
+    });
```

Comment:
I wonder if we can't create a slightly more true to life test. 

Assuming the marker is centered on the map, we should be able to click on the map (say at 0.1 .0.1) and see updated coordinates flow through.

Reply:
Would we do this by sending out a click event?

I don't believe there's a way to click on an element at a certain position using `HTMLElement.click()`
(https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click)[https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click]

Reply:
To query the selector for an interactive surface, and sub-subsequently click on it, I'll first need to render the map component (`baw-map`). However, this is impossible without removing existing safeguards that are present to prevent the embedded Google map from rendering in a testing environment

[source](https://github.com/QutEcoacoustics/workbench-client/blob/master/src/app/helpers/embedScript/embedGoogleMaps.ts#L14-L18)
![image](https://user-images.githubusercontent.com/33742269/206232303-e30d2218-b0cb-4a57-853b-3da89dc3c979.png)

From what I can see, there are two solutions:
- Disable the safeguards that prevent the embedded google maps from rendering in a testing environment (dangerous but more rigerous)
- Add another unit test for an emitted click event on the model that fakes the user clicking on the map and the model emitting an event similar to `domEvent: new Event("click")` (safer, but will use an abstracted / higher level test that doesn't test the embedded map)

Reply:
hmm let's try and find out why rendering googlemaps is bad in tests?

Reply:
Otherwise yeah, let's go with the method you had, where you simulate an event fired by a maps component

Reply:
I can't seem to find a reference to why it was implemented in PR #1972 (commit)[https://github.com/QutEcoacoustics/workbench-client/commit/07e18bc639888b4a347fd84dc951463a2d525fea#diff-f558660b3435ca693715d2962016a8ffb2f0f267167bc9c04c5974b5426e9637R14-R18]

My assumption is that we don't want to make an API call (with our key) every time we run tests as we:
- Don't want to bombard the Google Maps API with 10+ requests in fast succession when running unit tests
- Don't want to have unit tests potentially failing because of a. a bad internet connection, or b. because the maps API is unreachable. (Our unit tests don't want to be testing if Google is alive)
- We pay per request

(Similar (but not very friendly) response to a similar question on SO)[https://stackoverflow.com/questions/13657965/angularjs-unit-testing-application-based-on-google-maps-api#comment33737860_13657965]

Reply:
ok, in dev, record the event, and in tests simulate the firing, that should be enough

it still doesn't really test that we've bound to google's click handler but it's close enough

Reply:
I've left a JsDoc comment stating the need for manual testing

https://github.com/QutEcoacoustics/workbench-client/pull/2014/files#diff-ec8b1ef587154c95ecb3ba88fb4d1296a839d2fe11521acdb88174dffb0a7d97R11-R15